src/arrays/ladderops.jl

```diff
@@ -22,10 +22,22 @@
     before_eye(lens, pivot) = speye(prod(lens[[1:pivot-1]]))
     after_eye(lens, pivot) = speye(prod(lens[[pivot+1:length(lens)]]))
     function eye_sandwich(lens, pivot, op)
-        return kron(before_eye(lens, pivot), 
-                    op, 
+        return kron(before_eye(lens, pivot),
+                    op,
                     after_eye(lens, pivot))
     end
 
+function positionop(n::Int)
+    cop = raiseop(n)
+    return (cop+cop')/sqrt(2.)
+end
+
+function momentumop(n::Int)
+    cop = raiseop(n)
+    return im*(cop-cop')/sqrt(2.)
+end
+
```

Comment:
A couple of points here:

1. We should support define these operator functions so that they can target specific factors of product states. See how `raiseop` and `lowerop` for an example of this; you could just reuse `eye_sandwhich` to make it work. Eventually, though, we should replace the usage of that method with specialized implementations that don't make temporary arrays. 

2. I believe it would be more efficient to scale the resulting `QuArray`s in place than make a copy, e.g.
```julia
function momentumop(n::Int)
    cop = raiseop(n)
    return scale!(im/sqrt(2.), cop-cop')
end
```